integrated_channels/sap_success_factors/utils.py

```diff
@@ -32,45 +36,91 @@ def get_serialized_data(self):
         }
         return json.dumps(final_structure, sort_keys=True).encode('utf-8')
 
+    def resolve_removed_courses(self, previous_audit_summary):
+        """
+        Ensures courses that are no longer in the catalog get properly marked as inactive.
+
+        Args:
+            previous_audit_summary (dict): The previous audit summary from the last course export.
+
+        Returns:
+            An audit summary of courses with information about their presence in the catalog and current status.
+        """
+        if self.removed_courses_resolved:
+            return {}
+
+        new_audit_summary = {}
+
+        for course in self.courses:
+            key = course['courseID']
+            status = course['status']
+            new_audit_summary[key] = {
+                'in_catalog': True,
+                'status': status,
+            }
+            # Remove the key from previous audit summary so we can process courses that are no longer present.
+            if key in previous_audit_summary:
+                del previous_audit_summary[key]
+
+        for key, summary in previous_audit_summary.items():
+            new_audit_summary[key] = {
```

Comment:
Do we want to add this unconditionally? Or only if the course has not previously been set as inactive? It seems like this would carry courses forward in the audit indefinitely. I would think that, when removing inactive courses, we'd only want them to show up in the audit log of the run we actually used to perform the removal.